Add stan extension

jrnold opened the pull request:
Add support to recognize Stan model files (extension `.stan`). [Stan](https://github.com/stan-dev/stan/) is a probabilistic programming language. The extension `.stan` does not conflict with any existing extensions in `languages.yml`. Here is the relevant search: https://github.com/search?utf8=%E2%9C%93&q=extension%3Astan+NOT+nothack&type=Code&ref=searchresults

pchaigno commented on lib/linguist/languages.yml:
```diff
@@ -3355,6 +3355,13 @@ Squirrel:
   ace_mode: c_cpp
 
+Stan:
+  type: programming
```
We should choose a color for this language (it will be displayed in the language statistics bar). Is there an official logo from which we could extract a color?

pchaigno commented:
Nice pull request :+1:

I found this [Stan Sublime Text package](https://github.com/dougalsutherland/sublime-stan) for syntax highlighting. Could you check that it's good enough? (You can use [Lightshow](https://github-lightshow.herokuapp.com/) if you don't have Sublime Text).

jrnold commented:
I added a color for Stan, which is the brick red color of the logo (no other languages use that exact color) ![Stan logo](https://raw.githubusercontent.com/stan-dev/logos/master/logo.png)

I'll take a look at the Stan Sublime Text package; I think it looks good, though it's missing a couple things from a few newer versions of the Stan modeling language. My plan had been to submit this, and then go back and submit a pull request to for the Sublime Text package to get it updated and/or write an Atom package.

pchaigno commented:
If the Sublime Text package is good enough, I think we should add it now. Then, if it's updated to support newer versions of Stan, Linguist will automatically pull the changes. If you write a better Atom package, we can always change it to use that package.

jrnold commented:
The Sublime Text package looks good.

jrnold commented:
Hey @dougalsutherland : We are adding support to github for recognizing and highlighting Stan files. Since linguist can use Sublime Text grammars for highlighting we were going to use your [Sublime Stan](https://github.com/dougalsutherland/sublime-stan) for the highlighter. I wanted to give you a heads up and see if you're still maintaining it.

jrnold commented:
I added my own grammar from a new Atom package (based off the Sublime Stan grammar). It's at https://github.com/jrnold/atom-language-stan, although I added it as submodule and to grammars.yml

jrnold commented:
To @syclik This pull request should get Stan recognized by the github language detector and syntax highlighter.

arfon commented:
Nice Pull Request. Thanks @jrnold!

This will be live in the next release of Linguist (this week sometime)

dougalsutherland commented:
@jrnold Whoops, missed that notification somehow. I'm not actively maintaining sublime-stan anymore (don't really use Stan these days), but I might look into updating it to use https://github.com/jrnold/stan-language-definitions since that would be much nicer.

jrnold commented:
@dougalsutherland No problem. I ended up writing an Atom package for Stan (https://github.com/jrnold/atom-language-stan). I think the grammar for Atom is completely compatible with Sublime and TextMate, just written in CSON. So you can convert CSON -> JSON -> plist if need be. I started with your grammar, but ended up making a few changes. The largest is that it no longer checks that only blocks or comments appear at the top level: I did that so that the grammar could be used to highlight fragments of Stan models. I linter can handle checking the syntax of the model.
